File: 26-crud-operation-using-asp.net-core-api-(.NET-7.0)/CrudOperation.API/CrudOperation.API/Models/Domain/Employee.cs
using System.ComponentModel.DataAnnotations;
using System.Numerics;

namespace CrudOperation.API.Models.Domain
{
    public class Employee
    {
        [Key]
        public Guid Id { get; set; }
        [StringLength(50)]
        public string FirstName { get; set; }
        [StringLength(50)]
        public string? Middlename { get; set; }
        [StringLength(50)]
        public string LastName { get; set; }
        [StringLength(50)]
        public string Email { get; set; }
        [StringLength(10)]
        public string Mobile { get; set; }
        [StringLength(50)]
        public string Company { get; set; }
        public long Salary { get; set; }
        [StringLength(50)]
        public string Country { get; set; }
        [StringLength(50)]
        public string City { get; set; }
        [StringLength(50)]
        public string State { get; set; }
    }
}

File: 26-crud-operation-using-asp.net-core-api-(.NET-7.0)/crud-operation-using-web-api/src/app/model/employee.ts
// export class Employee {

//     id:string;
//     firstName : string;
//     middleName : string;
//     lastName : string;
//     email:string;
//     mobile : string;
//     company : string;
//     salary : number;
//     country:string;
//     state:string;
//     city:string;
  
//     constructor(id: string, firstName: string, middleName: string, lastName: string,email:string,mobile:string,company:string,salary:number,country:string,state:string,city:string) {
//         this.id = id;
//         this.firstName = firstName;
//         this.middleName = middleName;
//         this.lastName=lastName;
//         this.email = email;
//         this.mobile = mobile;
//         this.company = company;
//         this.salary=salary;
//         this.country=country;
//         this.state=state;
//         this.city=city;
//     }
//   }

  export interface Employee {
    id:string;
    firstName : string;
    middleName : string;
    lastName : string;
    email:string;
    mobile : string;
    company : string;
    salary : number;
    country:string;
    state:string;
    city:string;
  }

  export interface Response {
    id:string;
    isSuccess : boolean;
    HttpStatusCode : string;
    Messages:Messages[]
  }

  export interface Messages {
    Message : string;
  }